parsimonious/utils.py

```diff
@@ -1,22 +1,15 @@
 """General tools which don't depend on other parts of Parsimonious"""
 
 import ast
-from sys import version_info
 
 from six import python_2_unicode_compatible
 
 
 class StrAndRepr(object):
-    """Mix-in to add a ``__str__`` and ``__repr__`` which return the
-    UTF-8-encoded value of ``__unicode__``"""
-
-    if version_info >= (3,):
-        # Don't return the "bytes" type from Python 3's __str__:
-        def __repr__(self):
-            return self.__str__()
-    else:
-        def __repr__(self):
-            return self.__str__().encode('utf-8')
```

Comment:
Yes, this was definitely wrong. The result of `__str__` should already have been encoded, being a str. Good catch!

It might even be nice to add a test for this (which you pretty much did in your commit message), though I wouldn't block the merge on it.